src/constants/analytics.ts

```diff
@@ -7,6 +7,36 @@ import { TransferNotificationTypes } from './notifications';
 import type { TradeTypes } from './trade';
 import type { DydxAddress, EvmAddress, WalletConnectionType, WalletType } from './wallets';
 
+type AnalyticsEventDataWithReferrer<T extends AnalyticsEvent> = AnalyticsEventData<T> & {
+  referrer: string;
+};
+export type AnalyticsEventTrackMeta<T extends AnalyticsEvent> = {
+  detail: {
+    eventType: AnalyticsEvent;
+    eventData?: AnalyticsEventDataWithReferrer<T>;
```

Comment:
should this be non-optional since there'll always be a `referrer` property (even though it might be an empty string?)

Reply:
that's a good point, will update